### DISCUSSION TITLE
DisableBatchPacketLimit on listener

### DISCUSSION THREAD

**Discussion by AsherMaximum** [Sandertv/gophertunnel#114] [Q&A]
My clients keep getting disconnected, and the message log shows "error reading from listener connection: number of packets 782 in compressed batch exceeds 768"

I can see that on conns from a Dial, the function DisableBatchPacketLimit() is called on the decoder when it is created.
For the listener, it is not.
I've worked around this issue by just changing the limit `maximumInBatch = 1024 + 1024` in decoder.go, but I'm hoping there's a way I can do this without keeping my own modified copy of gophertunnel.

Is there a way to increase or disable the packet limit for Listener conns?

**Comment by Sandertv** [comment]
Hi! Thanks for reporting this. This is an issue in gophertunnel and I will look into increasing the maximum amount of packets in a batch. This is an anti-DoS check. Do you have an estimate of the highest number you've see in this error message? Has it ever gone over 1000?

**Reply by AsherMaximum** [reply]
Yeah, it's gone above 1000, although I'm not sure how high. I changed it from the default of `512+256` to `512+512` first, and usually didn't hit it, but still hit it occasionally. 
Now I just brute forced it for testing, and added another 10000 to it.

I think when I was seeing the messages after I changed it to `512+512` I was seeing it as high as 3000, but I'm not certain.

**Reply by Sandertv** [reply]
Do you have any idea on what you have to do in-game to trigger that many packets?

**Reply by AsherMaximum** [reply]
The lower count happens upon login, usually hits about 800-900 packets. 
I forget what I was doing when it got well over 1000.
